src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -352,45 +377,54 @@ private AcceptTxResult AddCore(Transaction tx, TxFilteringState state, bool isPe
 
-                _transactions.TryGetBucketsWorstValue(tx.SenderAddress!, out Transaction? worstTx);
+                (tx.SupportsBlobs ? _blobTransactions : _transactions).TryGetBucketsWorstValue(tx.SenderAddress!, out Transaction? worstTx);
                 tx.GasBottleneck = (worstTx is null || effectiveGasPrice <= worstTx.GasBottleneck)
                     ? effectiveGasPrice
                     : worstTx.GasBottleneck;
 
-                bool inserted = _transactions.TryInsert(tx.Hash!, tx, out Transaction? removed);
-                if (inserted && tx.Hash != removed?.Hash)
-                {
-                    _transactions.UpdateGroup(tx.SenderAddress!, state.SenderAccount, UpdateBucketWithAddedTransaction);
-                    Metrics.PendingTransactionsAdded++;
-                    if (tx.Supports1559) { Metrics.Pending1559TransactionsAdded++; }
+                bool inserted = (tx.SupportsBlobs
+                    ? _blobTransactions.TryInsertBlobTx(tx, out Transaction? removed)
+                    : _transactions.TryInsert(tx.Hash!, tx, out removed));
 
-                    if (removed is not null)
-                    {
-                        EvictedPending?.Invoke(this, new TxEventArgs(removed));
-                        // transaction which was on last position in sorted TxPool and was deleted to give
-                        // a place for a newly added tx (with higher priority) is now removed from hashCache
-                        // to give it opportunity to come back to TxPool in the future, when fees drops
-                        _hashCache.DeleteFromLongTerm(removed.Hash!);
-                        Metrics.PendingTransactionsEvicted++;
-                    }
+                if (!inserted)
+                {
+                    Metrics.PendingTransactionsPassedFiltersButCannotReplace++;
+                    return AcceptTxResult.ReplacementNotAllowed;
                 }
-                else
+
+                if (tx.Hash == removed?.Hash)
                 {
-                    if (isPersistentBroadcast && inserted)
+                    if (isPersistentBroadcast)
                     {
                         // it means it was added and immediately evicted - we are adding only to persistent broadcast
                         _broadcaster.Broadcast(tx, isPersistentBroadcast);
                     }
                     Metrics.PendingTransactionsPassedFiltersButCannotCompeteOnFees++;
                     return AcceptTxResult.FeeTooLowToCompete;
                 }
+
+                (tx.SupportsBlobs ? _blobTransactions : _transactions).UpdateGroup(tx.SenderAddress!, state.SenderAccount, UpdateBucketWithAddedTransaction);
+                Metrics.PendingTransactionsAdded++;
+                if (tx.Supports1559) { Metrics.Pending1559TransactionsAdded++; }
+                if (tx.SupportsBlobs) { Metrics.PendingBlobTransactionsAdded++; }
```

Comment:
Would be better to delegate that to inner TxPools, so we don't have to have those if's

Reply:
Yeah, but we can't do it when inserting - later we have check for `tx.Hash == removed?.Hash` and we shouldn't increment metrics if tx is added and evicted right after - in practice it is not added. We can do it as override of `UpdateGroup`, but I don't like it, incrementation of metrics is not expected there. I can create new methods just for incrementing metrics, but it will overcomplicate simple thing. Current version is simpler and better readable